### PR TITLE
Add optimizations list

### PR DESCRIPTION
**Pull request by ColasPinson**
This PR adds information about optimizations run by the user as a list.

### CODE REVIEW

File: app/views/optimizations/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  </thead>
+
+  <tbody>
+    <% @optimizations.each do |optimization_request| %>
+    <tr>
+      <td>#<%= optimization_request.id %></td>
+      <td><%= optimization_request.owner.login %></td>
+      <td><%= optimization_request.kind %></td>
+      <td><%= optimization_request.outputs["status"] ? optimization_request.outputs["status"] : "Empty" %></td>
+      <td><%= optimization_request.inputs.empty? ? "0" : optimization_request.inputs["x"].length %></td>
```
Comment (relf) [inline app/views/optimizations/index.html.erb:24]: You should create a method `nb_of_evaluations` in the `Optimization` model

**Review by relf — CHANGES_REQUESTED** [review]
Ok that's pretty interesting

File: app/models/optimization.rb [hunk 2]
```diff
@@ -154,4 +154,18 @@ def log_error(err_msg)
   def raise_error(err_msg)
     raise OptimizationError.new(err_msg)
   end
+
+  def nb_points
+    return self.inputs.empty? ? "0" : self.inputs["x"].length
```
Comment (relf) [inline app/models/optimization.rb:159]: Hi Nicolas. You can remove `return`. In Ruby, the `return` keyword is optional when returning at the end of a function and in practice is only used when returning earlier.

Comment (ColasPinson) [reply app/models/optimization.rb:159]: Yes, I forgot, thank you for reminding me, I will fix this.

File: app/models/optimization.rb [hunk 3]
```diff
@@ -154,4 +154,34 @@ def log_error(err_msg)
   def raise_error(err_msg)
     raise OptimizationError.new(err_msg)
   end
+
+  def nb_points
+    self.inputs.empty? ? "0" : self.inputs["x"].length
+  end
+
+  def status_text
```
Comment (relf) [inline app/models/optimization.rb:162]: Actually this kind of methods are helpers for views and are rather located in a Ruby module under `app/helpers`.

File: test/fixtures/optimizations.yml [hunk 4]
```diff
@@ -1,3 +1,13 @@
 optim_ackley2d:
+  id: 2
   kind: SEGOMOE
-  config: <%= { :xlimits => [[-32.768, 32.768], [-32.768, 32.768]] }.to_json.inspect  %>
\ No newline at end of file
+  config: <%= { :xlimits => [[-32.768, 32.768], [-32.768, 32.768]] }.to_json.inspect  %>
+
+optim_unexpected_id:
+  id: -2.28
```
Comment (relf) [inline test/fixtures/optimizations.yml:7]: Indeed there is no such thing as an `id` being equal to `-2.28`! Not sure why you need that.

File: test/fixtures/optimizations.yml [hunk 5]
```diff
@@ -1,3 +1,13 @@
 optim_ackley2d:
+  id: 2
```
[the remark this answers is not in the document]
Comment (ColasPinson) [reply test/fixtures/optimizations.yml:2]: Good to know, thank you !

File: test/fixtures/optimizations.yml (hunk 4 above)
Comment (ColasPinson) [reply test/fixtures/optimizations.yml:7]: I tried to find every test i could think of, but if the ids are completely secure, no need to check for unexpected id, I agree.

**Review by relf — APPROVED** [review]

**Pull request merged**